deephub/trainer/exporters.py

```diff
@@ -0,0 +1,91 @@
+# Source: https://github.com/bluecamel/best_checkpoint_copier
+import glob
+import os
+import shutil
+import tensorflow as tf
+
+class Checkpoint(object):
+  dir = None
+  file = None
+  score = None
+  path = None
+
+  def __init__(self, path, score):
+    self.dir = os.path.dirname(path)
+    self.file = os.path.basename(path)
+    self.score = score
+    self.path = path
+
+
+class BestCheckpointCopier(tf.estimator.Exporter):
+  checkpoints = None
+  checkpoints_to_keep = None
+  compare_fn = None
+  name = None
+  score_metric = None
+  sort_key_fn = None
+  sort_reverse = None
+
+  def __init__(self, name='best_checkpoints', checkpoints_to_keep=5, score_metric='Loss/total_loss',
+               compare_fn=lambda x,y: x.score < y.score, sort_key_fn=lambda x: x.score, sort_reverse=False):
+    self.checkpoints = []
+    self.checkpoints_to_keep = checkpoints_to_keep
+    self.compare_fn = compare_fn
+    self.name = name
+    self.score_metric = score_metric
+    self.sort_key_fn = sort_key_fn
+    self.sort_reverse = sort_reverse
+    super(BestCheckpointCopier, self).__init__()
+
+  def _copyCheckpoint(self, checkpoint):
+    desination_dir = self._destinationDir(checkpoint)
+    os.makedirs(desination_dir, exist_ok=True)
+
```

Comment:
Maybe we should use shutil.copytree instead of this for loop?

Reply:
Or you want the logs? @nanton96

Reply:
I don't really mind losing the logs.

However, as we need to select only the files relating to the best checkpoint and ignore the rest using copytree with the ignore_patterns would be, in my opinion, less readable than what we have here...

Reply:
Ok, we can leave it as is then.